src/protocol-handlers/index.js

```diff
@@ -0,0 +1,138 @@
+const { app, shell, ipcMain } = require('electron')
+const i18n = require('i18next')
+const createToggler = require('../utils/create-toggler')
+const store = require('../common/store')
+const { showPrompt } = require('../dialogs')
+const setupI18n = require('../i18n')
+const { parseUrl, getGatewayUrl } = require('./urls')
+
+const CONFIG_KEY = 'askWhenOpeningIpfsURIs'
+const CONFIG_KEY_ACTION = 'openIpfsURIsAction'
+
+const ACTION_OPTIONS = {
+  OPEN_IN_BROWSER: 'openInBrowser',
+  OPEN_IN_IPFS_DESKTOP: 'openInIpfsDesktop'
+  // BROWSER_PUBLIC_GATEWAY: 'browserPublicGateway',
+  // BROWSER_LOCAL_GATEWAY: 'browserLocalGateway',
+  // FILES_SCREEN: 'filesScreen'
+  // EXPLORE_SCREEN: 'exploreScreen'
+}
+
+const DEFAULT_ACTION = ACTION_OPTIONS.OPEN_IN_BROWSER
+
+async function getAction () {
+  const ask = store.get(CONFIG_KEY, true)
+  if (!ask) {
+    return store.get(CONFIG_KEY_ACTION, DEFAULT_ACTION)
+  }
```

Comment:
I assume this is checking if we have a saved config preference, and if not, we're then returning the default action?

Does that mean that if we *do* have a config preference, that the below prompt shows? What am I missing?

Can we change `ask` to a more explicit name that explains what it's doing?

Reply:
This checks:
1. Am I supposed to ask when opening IPFS URIs? Default is yes.
2. Then, if you are not supposed to ask, it returns either the default action, or the user's preference.
Renamed to `askWhenOpeningUri`. Feel free to suggest something else.